### PR TITLE
Use vault.centos for centos-7 image repos

### PR DESCRIPTION
**Pull request by lukaszachy**
mirror/baseurl used in the image are gone now

Pull Request Checklist

* [x] implement the feature

### CODE REVIEW

File: containers/centos/7/Containerfile.upstream [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #
 
 FROM quay.io/centos/centos:7
+RUN cd /etc/yum.repos.d/ && sed '/mirrorlist/d' -i *repo && sed 's|#baseurl=http://mirror.centos.org/centos/$releasever|baseurl=https://vault.centos.org/7.9.2009|' -i *repo
```
Comment (happz) [inline containers/centos/7/Containerfile.upstream:8]: Please, add a comment why this sequence of commands is needed, and split it over multiple lines, it improves readability. See e.g. https://github.com/teemtee/tmt/blob/76b5ff2cf67661c8d603e3f7f12eaf4ea3742f6e/containers/fedora/rawhide/Containerfile.unprivileged#L11:

```docker
    # Populate dnf cache
RUN    dnf makecache \
    # Create unprivileged user and setup sudo for it
    && dnf install -y /usr/sbin/useradd \
    && useradd fedora \
    && usermod -aG wheel fedora \
    && echo -e 'fedora\tALL=(ALL)\tNOPASSWD: ALL' >> /etc/sudoers
```

Comment (lukaszachy) [reply containers/centos/7/Containerfile.upstream]: ack, done

**Review by happz — APPROVED** [review]

File: tests/prepare/require/test.sh [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
 
-    for image in fedora centos:7 ; do
+    for image in $TEST_IMAGE_PREFIX/fedora/40:latest \
+                 $TEST_IMAGE_PREFIX/centos/7/upstream:latest; do
         # Prepare the tmt command and expected error message
-        tmt="tmt run -avr provision -h container -i $image"
-        if [[ $image == fedora ]]; then
+        tmt="tmt run -avr provision -h $PROVISION_HOW -i $image"
+        if [[ $image =~ fedora ]]; then
             error='Unable to find a match: forest'
```
Comment (happz) [inline tests/prepare/require/test.sh:22]:
```suggestion
            error='No match for argument: forest'
```

**Review by KwisatzHaderach — APPROVED** [review]

**Comment by happz** [comment]
Unrelated failures with `virtual` plugin, merging.

**Pull request merged**